raft_test.go

```diff
@@ -983,6 +984,180 @@ func TestRaft_ApplyConcurrent_Timeout(t *testing.T) {
 	c.FailNowf("[ERR] Timeout waiting to detect apply timeouts")
 }
 
+func TestRaft_nextConfiguration_prevIndex(t *testing.T) {
+	// Stale prevIndex.
+	future := &configurationChangeFuture{
+		command:       AddStaging,
+		serverID:      ServerID("id1"),
+		serverAddress: ServerAddress("addr1"),
+		prevIndex:     1,
+	}
+	future.init()
+	next := nextConfiguration(Configuration{}, 2, future)
+	if len(next.Servers) > 0 || !future.responded ||
+		!strings.Contains(future.Error().Error(), "changed") {
+		t.Fatalf("nextConfiguration should have failed due to intervening configuration change")
+	}
+
+	// Current prevIndex.
```

Comment:
Ah I forgot we bypassed the promotion logic. These should probably be changed to `AddVoter` since these will start to fail later.